magentov2/plugin/Indodana/PayLater/Gateway/Http/Client/ClientMock.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Indodana\Paylater\Gateway\Http\Client;
+
+use Magento\Payment\Gateway\Http\ClientInterface;
+use Magento\Payment\Gateway\Http\TransferInterface;
+use Magento\Payment\Model\Method\Logger;
+
+class ClientMock implements ClientInterface
+{
+    const SUCCESS = 1;
+    const FAILURE = 0;
+
+    /**
+     * @var array
+     */
+    private $results = [
+        self::SUCCESS,
+        self::FAILURE
+    ];
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * @param Logger $logger
+     */
+    public function __construct(
+        Logger $logger
+    ) {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Places request to gateway. Returns result as ENV array
+     *
+     * @param TransferInterface $transferObject
+     * @return array
+     */
+    public function placeRequest(TransferInterface $transferObject)
+    {
+        //$this->logger->debug('test');
+
+        $response = $this->generateResponseForCode(
+            $this->getResultCode(
+                $transferObject
+            )
+        );
+
+        $this->logger->debug(
+            [
+                'request' => $transferObject->getBody(),
+                'response' => $response
+            ]
+        );
+
+        return $response;
+    }
+
+    /**
+     * Generates response
+     *
+     * @return array
+     */
+    protected function generateResponseForCode($resultCode)
+    {
+
+        return array_merge(
+            [
+                'RESULT_CODE' => $resultCode,
+                'TXN_ID' => $this->generateTxnId()
+            ],
+            $this->getFieldsBasedOnResponseType($resultCode)
+        );
+    }
+
+    /**
+     * @return string
+     */
+    protected function generateTxnId()
+    {
+        return md5(mt_rand(0, 1000));
```

Comment:
Any reason for randoming through 0-1000?

Reply:
no, it's just follow the sample